mqtt/mqtt-broker/src/broker.rs

```diff
@@ -651,7 +652,9 @@ where
                         }
                     } else {
                         info!("cleaning offline session for {}", client_id);
-                        let new_session = Session::new_transient(auth_id, connreq);
+                        let state =
+                            SessionState::new(client_id.clone(), self.config.session().clone());
```

Comment:
since we will not make separate settings per session, it may worthwhile to share config between sessions via Rc instead of cloning

Reply:
I think cloning session config is fine. For Rc<SessionConfig>, I would have to keep a filed in the broker and it does not give any big advantage.